code/modules/medical/genetics/geneticsMachines.dm

```diff
@@ -1761,88 +265,859 @@
 
 /obj/machinery/computer/genetics/proc/log_maybe_cheater(var/who, var/action = "")
 	// this is used repeatedly so let's just make it a proc and stop repeating ourselves 50 times
-	message_admins("[key_name(who)] [action] (failed href validation, maybe cheating)")
-	logTheThing("debug", who, null, "[action] but failed href validation.")
-	logTheThing("diary", who, null, "[action] but failed href validation.", "debug")
-
-
-/obj/machinery/computer/genetics/proc/encryption_challenge(var/bp_num, var/datum/basePair/bp, var/datum/bioEffect/E)
-	if(bp.marker == "locked")
-		boutput(usr, "<span class='notice'><b>SCANNER ALERT:</b> Encryption is a [E.lockedDiff]-character code.</span>")
-		var/characters = ""
-		for(var/X in E.lockedChars)
-			characters += "[X] "
-		boutput(usr, "<span class='notice'>Possible characters in this code: [characters]</span>")
-		if(genResearch.lock_breakers > 0)
-			boutput(usr, "<span class='notice'>[genResearch.lock_breakers] auto-decryptions available. Enter UNLOCK as the code to expend one.</span>")
-		var/code = input("Enter decryption code.","Genetic Decryption") as null|text
-		if(!code)
-			return
-		code = uppertext(code)
-		if (code == "UNLOCK")
-			if(genResearch.lock_breakers > 0)
-				genResearch.lock_breakers--
-				var/datum/basePair/bpc = E.dnaBlocks.blockList[bp_num]
-				bp.bpp1 = bpc.bpp1
-				bp.bpp2 = bpc.bpp2
-				bp.marker = "green"
-				bp.style = ""
-				boutput(usr, "<span class='notice'><b>SCANNER ALERT:</b> Base pair unlocked.</span>")
-				if (E.dnaBlocks.sequenceCorrect())
-					E.dnaBlocks.ChangeAllMarkers("white")
-				usr << link("byond://?src=\ref[src];viewpool=\ref[E]")
+	message_admins("[key_name(who)] [action] (failed validation, maybe cheating)")
+	logTheThing("debug", who, null, "[action] but failed validation.")
+	logTheThing("diary", who, null, "[action] but failed validation.", "debug")
+
+/obj/machinery/computer/genetics/ui_status(mob/user)
+	if (user in src.scanner)
+		return UI_UPDATE
```

Comment:
Note: behavior change

Reply:
This one actually isn't a behavior change. You were previously able to watch (but not interact with) the console while inside it but only if you opened the console before going into the pod.